src/storage/src/source/source_reader_pipeline.rs

```diff
@@ -382,6 +396,7 @@ where
                             yield Some(
                                 SourceReaderOperatorOutput {
                                     messages: std::mem::take(&mut untimestamped_messages),
+                                    status_update,
```

Comment:
why not `status_update.take()` here like below? Optimization because here we're exiting?

Reply:
I hadn't thought about it, but in retrospect it seems worthwhile to make them as similar as possible. Making that change.